kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -507,6 +507,10 @@ protected void validateWriteToExistingTable(Engine engine, SnapshotImpl snapshot
                 txnId.getAppId(), txnId.getVersion(), lastTxnVersion.get());
           }
         });
+    if (operation != Operation.REPLACE_TABLE && clusteringColumns.isPresent()
```

Comment:
I think the better way to do this check is to add `isCreateOrReplace` to the args here -- that's what we standardize on throughout the rest of the code